SoftwareDesign.md

```diff
@@ -0,0 +1,3 @@
+My favorite software design pattern is the State pattern because it's probably what React is modeled after. I personally enjoy using React and React Native so I have experienced the usefulness of it. Changing the object based on the state makes sense since it's something that is in line with how a user interacts with a website. If a user clicks on a button, then the state changes for that button changes, which is reflected in the button's components code. 
```

Comment:
https://reactjs.org/docs/design-principles.html
found this interesting trying to find if it's modeled after that.